### DISCUSSION TITLE
Use Ports Panel or Port Forwarding?

### DISCUSSION THREAD

**Discussion by jsjoeio** [microsoft/vscode-discussions#174] [Extension Development QnA] [answered]
Is there a way to leverage the native ports view or port forwarding within a VS Code Extension?

I did find a proposed API here: https://github.com/microsoft/vscode/blob/main/src/vscode-dts/vscode.proposed.tunnels.d.ts#L44

**Comment by jsjoeio** [comment]
Looks like this proposed API was removed in favor of a tunnels API. Not sure what that means for implementing this 🤔

https://github.com/microsoft/vscode/pull/163187

**Reply by isidorn** [reply]
I think this is just a direction we decided to take.
fyi @alexr00

**Comment by alexr00** [comment, marked as answer]
We don't currently plan to finalize the `openTunnel` API. The existing `asExternalUri` API will also cause forwarded ports to show in the Ports view.
